src/xrCore/Crypto/xr_sha.cpp

```diff
@@ -1,24 +1,32 @@
 #include "stdafx.h"
 #include "xr_sha.h"
 
-namespace crypto
-{
+using namespace crypto;
 
-const xr_sha1::sha_checksum_t& xr_sha1::calculate(const u8* data, u32 data_size)
+const xr_sha1::sha_checksum_t& xr_sha1::calculate(const u8* data, u32 data_size, std::optional<sha_process_yielder> yielder)
```

Comment:
Why not take a pointer to `sha_process_yielder` here? `std::optional` is good for returning an optional value, but in this case it's rather redundant.

Reply:
The main purpose is to rid of raw pointers that can become dangling any moment.
Yes, std::optional creates a copy of yielder on its creation, but it is safer.
I can return it to a pointer if you want.

Also there is no need in 2 calculate functions. `yielder` could be a default parameter.